server/src/main/java/org/elasticsearch/action/search/OpenPointInTimeResponse.java

```diff
@@ -18,22 +19,46 @@
 import java.util.Base64;
 import java.util.Objects;
 
+import static org.elasticsearch.rest.action.RestActions.buildBroadcastShardsHeader;
+
 public final class OpenPointInTimeResponse extends ActionResponse implements ToXContentObject {
     private final BytesReference pointInTimeId;
 
-    public OpenPointInTimeResponse(BytesReference pointInTimeId) {
+    private final int totalShards;
+    private final int successfulShards;
+    private final int failedShards;
+    private final int skippedShards;
+
+    public OpenPointInTimeResponse(
+        BytesReference pointInTimeId,
+        int totalShards,
+        int successfulShards,
+        int failedShards,
+        int skippedShards
+    ) {
         this.pointInTimeId = Objects.requireNonNull(pointInTimeId, "Point in time parameter must be not null");
+        this.totalShards = totalShards;
+        this.successfulShards = successfulShards;
+        this.failedShards = failedShards;
+        this.skippedShards = skippedShards;
     }
 
     @Override
     public void writeTo(StreamOutput out) throws IOException {
         out.writeBytesReference(pointInTimeId);
+        if (out.getTransportVersion().onOrAfter(TransportVersions.ALLOW_PARTIAL_SEARCH_RESULTS_IN_PIT)) {
+            out.writeVInt(totalShards);
+            out.writeVInt(successfulShards);
+            out.writeVInt(failedShards);
+            out.writeVInt(skippedShards);
+        }
     }
 
     @Override
     public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
         builder.startObject();
         builder.field("id", Base64.getUrlEncoder().encodeToString(BytesReference.toBytes(pointInTimeId)));
+        buildBroadcastShardsHeader(builder, params, totalShards, successfulShards, failedShards, skippedShards, null);
```

Comment:
Not sure about this one.. We're changing the default output even for PIT requests that don't specify `allow_empty_search_results` . Should I wrap all these shard stats in a record and add them to the response only when `allow_partial_search_results` is explicitly specified?